### PR TITLE
Add immutability supporting limitation in documents.

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: site/content/docs/main/backup-reference.md [hunk 1]
```diff
@@ -74,6 +76,8 @@ Please do notice there is also side effect that may not be expected. Because sch
 
 If there is possibility the schedule will be disable to not create backup anymore, and the created backups are still useful. Please do not enable this option. For detail, please reference to [Backups created by a schedule with useOwnerReferenceInBackup set do not get synced properly](https://github.com/vmware-tanzu/velero/issues/4093).
 
+#### Cannot support bucket's RetetionPolicy
+From Velero's release-1.11 branch, Velero cannot support BSL's bucket with RetentionPolicy setting. Because Velero introduced a `Finalizing` state, Velero will first enter `Finalizing` phase after backup completion, and then the Finalizing controller will check whether there is any async operation in progress. If there is, then Velero will wait for all async operations complete, then turn backup into `Complete` phase. If there is not, then Velero will turn into `Complete` directly. By current design, whether there is async operation included or not, Velero's metadata in the OSS bucket need to change after it's created. This is the reason why RetentionPolicy cannot be supported.
```
[the remark this answers is not in the document]
Comment (blackpiglet) [reply site/content/docs/main/backup-reference.md:80]: Thanks for the suggestion.
I will reword the limitation to a more generic phrase, and add the Azure, AWS, and GCP document links.

**Comment by blackpiglet** [comment]
This address the need to document the data immutability limitation in issue #6422.

**Comment by blackpiglet** [comment]
@draghuram 
Could you help PTAL?

**Review by danfengliu — DISMISSED** [review]

**Comment by draghuram** [comment]
Sorry for the delay, will take a look.

**Comment by draghuram** [comment]
I made some changes to the text. Please feel free to make more changes as you see fit. It is a very difficult topic to document but it is better to have something and then we can improve it. 

### My version

Starting from 1.11, Velero's backups may not work as expected when the target object storage has some kind of an "immutability" option configured. These options are known by different names (see links below for some examples). The main reason is that Velero first saves the state of a backup as `Finalizing` and then checks whether there are any async operations in progress. If there are, it needs to wait for all of them to be finished before moving the backup state to `Complete`. If there are no async operations, the state is moved to `Complete` right away. In either case, Velero needs to modify the metadata in object storage and that will not be possible if some kind of immutability is configured on the object storage.

Note that backups may still work in some cases depending on specific provider and configuration. 

* For AWS S3 service, backups work because S3's object lock only applies to versioned buckets and the object data can still be updated as the new version. But when backups are deleted, old versions of the objects will not be deleted.
* Azure Storage Blob supports both versioned-level immutability and container-level immutability. For the versioned-level scenario, data immutability can still work in Velero, but the container-level cannot.
* GCP Cloud storage policy only supports bucket-level immutability, so there is no way to make it work in the GCP environment.

The following are the links to cloud providers' documentation in this regard:
* [AWS S3 Using S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html)
* [Azure Storage Blob Containers - Lock Immutability Policy](https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-policy-configure-version-scope?tabs=azure-portal)
* [GCP cloud storage Retention policies and retention policy locks](https://cloud.google.com/storage/docs/bucket-lock)

**Comment by blackpiglet** [comment]
@draghuram 
Thanks a lot! That looks better. 
Just to add some information from Velero's design perspective.

File: site/content/docs/main/backup-reference.md [hunk 2]
```diff
@@ -74,7 +76,23 @@ Please do notice there is also side effect that may not be expected. Because sch
...
 
+#### Cannot support backup data immutability
+Starting from 1.11, Velero's backups may not work as expected when the target object storage has some kind of an "immutability" option configured. These options are known by different names (see links below for some examples). The main reason is that Velero first saves the state of a backup as Finalizing and then checks whether there are any async operations in progress. If there are, it needs to wait for all of them to be finished before moving the backup state to Complete. If there are no async operations, the state is moved to Complete right away. In either case, Velero needs to modify the metadata in object storage and that will not be possible if some kind of immutability is configured on the object storage.
+
+Please also notice, before 1.11, even Velero can work with the target object storage with an "immutability" configuration, Velero didn't provide any "immutability" options. It can work because there is no limitation, but no official support in the data "immutability" function too.
```
Comment (draghuram) [inline site/content/docs/main/backup-reference.md:82]: This paragraph is a bit confusing to me. Do you mean to say that Velero pre-1.11 will work with immutability option though there will be some issues (such as old versions not being deleted)?

Comment (blackpiglet) [reply site/content/docs/main/backup-reference.md:82]: No. I mean to say that cannot supporting immutability is a new introducing issue, but Velero doesn't provide any support on this before.

Comment (draghuram) [reply site/content/docs/main/backup-reference.md:82]: OK. In that case, I would suggest this:

"Even with versions prior to 1.11, there was no explicit support in Velero to work with object storage that has  "immutability" configuration. As a result, you may see some problems even though backups seem to work (e.g. versions objects not being deleted when backup is deleted)."

Comment (blackpiglet) [reply site/content/docs/main/backup-reference.md:82]: Thanks a lot for the rephrasing.

**Review by Lyndon-Li — APPROVED** [review]

**Review by danfengliu — APPROVED** [review]

**Pull request merged**